storage/country_tree_facade.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "storage/country_tree.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/unordered_map.hpp"
+
+template <class K>
+struct CountryTreeKeyHasher
+{
+  size_t operator()(K const & k) const { return hash<string>()(k.Name()); }
+};
+
+template<>
+struct CountryTreeKeyHasher<int>
+{
+  size_t operator()(int k) const { return hash<int>()(k); }
+};
+
+/// This class is developed for using in Storage. It's a implementation of a tree.
+/// It should be filled with AddAtDepth method.
+/// This class is used in Storage and filled based on countries.txt (countries_migrate.txt).
+/// While filling CountryTree nodes in countries.txt should be visited in DFS order.
+template <class T>
+class CountryTreeFacade
```

Comment:
I removed the most part of unused CountryTreeFacade method. So now CountryTreeFacade and CountryTree have mostly different interface. Probably it's time to thing about renaming CountryTreeFacade to CountryTreeContainer.